File: 35. Search Insert Position/binary search solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let start = 0;
    let end = nums.length - 1;
    let mid;
    let ans = nums.length;
    while(start <= end) {
        mid = Math.floor(start + (end - start)/2);
        if(nums[mid] >= target) {
            ans = mid;
            end = mid - 1;
        } else {
            start = mid + 1;
        }
    }
    return ans;
};

File: 35. Search Insert Position/binary search solution.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int start = 0;
        int n = nums.size();
        int end = n - 1;
        int mid;
        int ans = n;
        while(start <= end) {
            mid = start + ((end - start)/2);
            if(nums[mid] >= target) {
                ans = mid;
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return ans;
    }
};
